skdownscale/pointwise_models/groupers.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import pandas as pd
+
+
+class SkdownscaleGroupGeneratorBase:
+    pass
+
+
+class PaddedDOYGrouper(SkdownscaleGroupGeneratorBase):
+    def __init__(self, df, offset=15):
+        self.df = df
+        self.offset = offset
+        self.max = 365
+        self.days_of_year = np.arange(1, 366)
+        self.days_of_year_wrapped = np.pad(self.days_of_year, 15, mode="wrap")
+        self.n = 1
+
+    def __iter__(self):
+        self.n = 1
+        return self
+
+    def __next__(self):
+        # n as day of year
+        if self.n > self.max:
+            raise StopIteration
+
+        i = self.n - 1
+        total_days = (2 * self.offset) + 1
+
+        # create day groups with +/- days
+        # number of days defined by offset
+        first_half = self.days_of_year_wrapped[i : i + self.offset]
+        sec_half = self.days_of_year_wrapped[self.n + self.offset : i + total_days]
+        all_days = np.concatenate((first_half, np.array([self.n]), sec_half), axis=0)
+
+        assert len(set(all_days)) == total_days, all_days
+
+        result = self.df[self.df.index.dayofyear.isin(all_days)]
+
+        self.n += 1
+
+        return self.n - 1, result
+
+    def mean(self):
+        list_result = []
+        for key, group in self:
+            list_result.append(group.mean().values[0])
+        result = pd.Series(list_result, index=self.days_of_year)
```

Comment:
shouldn't this be a DataFrame. In Pandas, `df.groupby(lambda x: x.month).mean() -> pd.DataFrame` if `df` is a DataFrame.

Reply:
I think it should actually be a `Series`? It seems like if I make it a DataFrame, the time index gets wonky

Reply:
ahh nm, got this to work in conjunction along with some other updates for your comment below